Create template from codelens doesn't work in IDEA EAP 2023-3 EAP

Click on create codelens template in Java file does nothing in IDEA EAP 2023-3 EAP

![image](https://github.com/redhat-developer/intellij-quarkus/assets/1932211/6b543288-233f-45b2-a385-e60a42f0d57d)

The problem is because LSP request doesn't contains as parameters the resolved class type (sourceType)

```json
[Trace - 19:17:04] Sending request 'workspace/executeCommand - (26)'.
Params: {
  "command": "qute.command.generate.template.content",
  "arguments": [
    {
      "projectUri": "renotes",
      "templateFileUri": "file:/C:/Users/azerr/git/renotes/src/main/resources/templates/Notes/newNotes.html",
      "templateFilePath": "src/main/resources/templates/Notes/newNotes.html",
      "parameters": [
        {
          "key": "notes",
          "sourceType": "List\u003cNote\u003e"
        },
        {
          "key": "currentNote",
          "sourceType": "Note"
        }
      ]
    }
  ]
}

```

For instance  `"sourceType": "Note"` although it should be  `"sourceType": "model.Note"`

After clickon codelens, there is an error reponse:

```json
[Trace - 19:17:04] Received response 'workspace/executeCommand - (26)' in 12ms.
No result returned.
Error: {
  "code": -32603,
  "message": "Internal error.",
  "data": "java.util.concurrent.CompletionException: io.quarkus.qute.TemplateException\r\n\tat java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)\r\n\tat java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:687)\r\n\tat java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:662)\r\n\tat java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2168)\r\n\tat com.redhat.qute.services.commands.QuteGenerateTemplateContentCommandHandler.executeCommand(QuteGenerateTemplateContentCommandHandler.java:68)\r\n\tat com.redhat.qute.ls.QuteWorkspaceService.lambda$executeCommand$0(QuteWorkspaceService.java:73)\r\n\tat com.redhat.qute.ls.QuteWorkspaceService.lambda$computeAsync$1(QuteWorkspaceService.java:105)\r\n\tat java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)\r\n\tat java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)\r\n\tat java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)\r\n\tat java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)\r\n\tat java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)\r\n\tat java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)\r\n\tat java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)\r\nCaused by: io.quarkus.qute.TemplateException\r\n\tat io.quarkus.qute.CompletedStage.get(CompletedStage.java:48)\r\n\tat io.quarkus.qute.MultiResultNode.process(MultiResultNode.java:20)\r\n\tat io.quarkus.qute.MultiResultNode.process(MultiResultNode.java:20)\r\n\tat io.quarkus.qute.MultiResultNode.process(MultiResultNode.java:20)\r\n\tat io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.lambda$renderData$5(TemplateImpl.java:239)\r\n\tat io.quarkus.qute.CompletedStage.whenComplete(CompletedStage.java:268)\r\n\tat io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.renderData(TemplateImpl.java:233)\r\n\tat io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.renderAsyncNoTimeout(TemplateImpl.java:224)\r\n\tat io.quarkus.qute.TemplateImpl$TemplateInstanceImpl.render(TemplateImpl.java:149)\r\n\tat com.redhat.qute.services.commands.QuteGenerateTemplateContentCommandHandler.lambda$executeCommand$0(QuteGenerateTemplateContentCommandHandler.java:73)\r\n\tat java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:684)\r\n\t... 12 more\r\nCaused by: io.quarkus.qute.TemplateException: Rendering error: Property \"isIterable\" not found on the base object \"null\" in expression {class.resolvedJavaType.isIterable}\r\n\tat io.quarkus.qute.TemplateException$Builder.build(TemplateException.java:169)\r\n\tat io.quarkus.qute.EvaluatorImpl.propertyNotFound(EvaluatorImpl.java:247)\r\n\tat io.quarkus.qute.EvaluatorImpl.resolve(EvaluatorImpl.java:201)\r\n\tat io.quarkus.qute.EvaluatorImpl.resolveReference(EvaluatorImpl.java:128)\r\n\tat io.quarkus.qute.EvaluatorImpl.lambda$resolveReference$2(EvaluatorImpl.java:132)\r\n\tat io.quarkus.qute.CompletedStage.thenCompose(CompletedStage.java:232)\r\n\tat io.quarkus.qute.EvaluatorImpl.resolveReference(EvaluatorImpl.java:132)\r\n\tat io.quarkus.qute.EvaluatorImpl.lambda$resolveReference$2(EvaluatorImpl.java:132)\r\n\tat io.quarkus.qute.CompletedStage.thenCompose(CompletedStage.java:232)\r\n\tat io.quarkus.qute.EvaluatorImpl.resolveReference(EvaluatorImpl.java:132)\r\n\tat io.quarkus.qute.EvaluatorImpl.evaluate(EvaluatorImpl.java:87)\r\n\tat io.quarkus.qute.ResolutionContextImpl$ChildResolutionContext.evaluate(ResolutionContextImpl.java:87)\r\n\tat io.quarkus.qute.IfSectionHelper$OperandCondition.evaluate(IfSectionHelper.java:249)\r\n\tat io.quarkus.qute.IfSectionHelper$DoubletonContext.resolve(IfSectionHelper.java:87)\r\n\tat io.quarkus.qute.IfSectionHelper.resolve(IfSectionHelper.java:45)\r\n\tat io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)\r\n\tat io.quarkus.qute.SectionNode.resolve(SectionNode.java:58)\r\n\tat io.quarkus.qute.SectionNode$SectionResolutionContextImpl.execute(SectionNode.java:228)\r\n\tat io.quarkus.qute.SectionHelper$SectionResolutionContext.execute(SectionHelper.java:76)\r\n\tat io.quarkus.qute.LoopSectionHelper.nextElement(LoopSectionHelper.java:127)\r\n\tat io.quarkus.qute.LoopSectionHelper.lambda$resolve$0(LoopSectionHelper.java:56)\r\n\tat io.quarkus.qute.CompletedStage.thenCompose(CompletedStage.java:232)\r\n\tat io.quarkus.qute.LoopSectionHelper.resolve(LoopSectionHelper.java:45)\r\n\tat io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)\r\n\tat io.quarkus.qute.SectionNode.resolve(SectionNode.java:58)\r\n\tat io.quarkus.qute.SectionNode$SectionResolutionContextImpl.execute(SectionNode.java:228)\r\n\tat io.quarkus.qute.SectionHelper$SectionResolutionContext.execute(SectionHelper.java:66)\r\n\tat io.quarkus.qute.Parser$1.resolve(Parser.java:1274)\r\n\tat io.quarkus.qute.SectionNode.resolve(SectionNode.java:53)\r\n\tat io.quarkus.qute.SectionNode.resolve(SectionNode.java:58)\r\n\t... 17 more\r\n"
}

```

--- a/src/main/java/com/redhat/devtools/intellij/qute/psi/internal/java/QuteJavaCodeLensCollector.java
+++ b/src/main/java/com/redhat/devtools/intellij/qute/psi/internal/java/QuteJavaCodeLensCollector.java
@@ -19,6 +19,7 @@ import com.redhat.devtools.intellij.lsp4ij.LSPIJUtils;
 import com.redhat.devtools.intellij.lsp4mp4ij.psi.core.utils.IPsiUtils;
 import com.redhat.devtools.intellij.qute.psi.QuteCommandConstants;
 import com.redhat.devtools.intellij.qute.psi.utils.PsiQuteProjectUtils;
+import com.redhat.devtools.intellij.qute.psi.utils.PsiTypeUtils;
 import com.redhat.devtools.intellij.qute.psi.utils.TemplatePathInfo;
 import com.redhat.qute.commons.datamodel.DataModelParameter;
 import com.redhat.qute.commons.datamodel.GenerateTemplateInfo;
@@ -109,7 +110,7 @@ public class QuteJavaCodeLensCollector extends AbstractQuteTemplateLinkCollector
             //ITypeBinding binding = parameterType.resolveBinding();
             DataModelParameter parameter = new DataModelParameter();
             parameter.setKey(parameterName);
-            parameter.setSourceType(parameterType.getPresentableText());
+            parameter.setSourceType(PsiTypeUtils.resolveSignature(parameterType, false));
             parameters.add(parameter);
         }
         return parameters;

